src/frontend/src/handler/explain_analyze_stream_job.rs

```diff
@@ -0,0 +1,430 @@
+    }
+
+    // find root node, and fill in dispatcher edges + nodes.
+    let root_or_dispatch_nodes = find_root_nodes(&operator_id_to_stream_node);
+    let mut root_node = None;
+    for operator_id in root_or_dispatch_nodes {
+        let node = operator_id_to_stream_node.get_mut(&operator_id).unwrap();
+        let fragment_id = node.fragment_id;
+        if let Some(merge_operator_id) = fragment_id_to_merge_operator_id.get(&fragment_id) {
+            let mut dispatcher = StreamNode::new_for_dispatcher(fragment_id);
+            dispatcher.dependencies.push(operator_id);
+            assert!(
+                operator_id_to_stream_node
+                    .insert(fragment_id as _, dispatcher)
+                    .is_none()
+            );
+            operator_id_to_stream_node
+                .get_mut(merge_operator_id)
+                .unwrap()
+                .dependencies
+                .push(fragment_id as _);
+        } else {
+            root_node = Some(operator_id);
+        }
+    }
+
+    (root_node.unwrap(), operator_id_to_stream_node)
+}
+
+fn find_root_nodes(stream_nodes: &HashMap<u64, StreamNode>) -> HashSet<u64> {
+    let mut all_nodes = stream_nodes.keys().copied().collect::<HashSet<_>>();
+    for node in stream_nodes.values() {
+        for dependency in &node.dependencies {
+            all_nodes.remove(dependency);
+        }
+    }
+    all_nodes
+}
+
+// Do a DFS based rendering. Each node will occupy its own row.
+// Schema:
+// | Operator ID | Identity | Actor IDs | Metrics ... |
+// Each node will be indented based on its depth in the graph.
+fn render_graph_with_metrics(
+    adjacency_list: &HashMap<u64, StreamNode>,
+    root_node: u64,
+    stats: &StreamNodeStats,
+    profiling_duration: &Duration,
+) -> Vec<ExplainAnalyzeStreamJobOutput> {
+    let profiling_duration_secs = profiling_duration.as_secs_f64();
+    let mut rows = vec![];
+    let mut stack = vec![(String::new(), true, root_node)];
+    while let Some((prefix, last_child, node_id)) = stack.pop() {
+        let Some(node) = adjacency_list.get(&node_id) else {
+            continue;
+        };
+        let is_root = node_id == root_node;
+
+        let identity_rendered = if is_root {
+            node.identity.clone()
+        } else {
+            let connector = if last_child { "└─ " } else { "├─ " };
+            format!("{}{}{}", prefix, connector, node.identity)
+        };
+
+        let child_prefix = if is_root {
+            ""
+        } else if last_child {
+            "   "
+        } else {
+            "│  "
+        };
+        let child_prefix = format!("{}{}", prefix, child_prefix);
+
+        let stats = stats.inner.get(&node_id);
+        let (output_throughput, output_latency) = stats
+            .map(|stats| (stats.total_output_throughput, stats.total_output_pending_ms))
+            .unwrap_or((0, 0));
+        let row = ExplainAnalyzeStreamJobOutput {
+            operator_id: node.operator_id.to_string(),
```

Comment:
I think we can hide the operator_id to users, since it is not a user-facing message.

Reply:
fixed https://github.com/risingwavelabs/risingwave/commit/cc4bc13811268cf3f6e3e883dfd22bfd001dd53a